src/dbus_fast/proxy_object.py

```diff
@@ -72,6 +83,22 @@ def _check_method_return(msg, signature=None):
                 msg,
             )
 
+    @staticmethod
+    def remove_signature(data: Any):
```

Comment:
I think it would be better/safer if this was a pure function and didn't modify `data`.

Reply:
What is the reason for making this a method on the `BaseProxyInterface` class? It seems like it could just be a standalone function and it doesn't seem to be particularly related to proxies.

Reply:
So in truth I was just moving it from here:
https://github.com/home-assistant/supervisor/blob/b8249548aef05be3353fb6dc45b7df4a200bfa53/supervisor/utils/dbus.py#L56-L71

I'm fine with making it a standalone function. As for making it functional and not modify data though, that's a trickier. It absolutely makes sense but that means making a deep copy and @bdraco and I were worried about the performance hit there.

Reply:
If there is any kind of volume we should avoid the deepcopy. If its not high volume it won't matter

Reply:
The function already makes a copy of lists and reassigns every key in dictionaries, so it seems like it is already doing as much work as a deep copy. And there is a separate `deepcopy` in one of the calls I saw, so in that case, it would be doing the equivalent work of two deep copies which could be reduced to 1 if this was a pure function.

Reply:
Will generate a profile to see how much its costing in bleak

Reply:
Its about 4.1% of the run time of processing incoming messages in bleak.  The `isinstance` checks and self calling look like they are more expensive than the actual copying

![unpack_variants](https://user-images.githubusercontent.com/663432/190205185-9310aead-8210-4cbf-9b82-75b1d736647a.png)

Reply:
Fair enough. I'll make the change then.

Reply:
https://github.com/hbldh/bleak/blob/d343cdd96cf6eb3de19d1dc6a85af664a96bc36d/bleak/backends/bluezdbus/utils.py#L30

Reply:
The added deepcopy was only for signals for reference. That was mainly because I noticed that if you have multiple signal handlers for the same signal and one of the handlers does modify data it actually affects the other handler. Since the same object is passed into each. So in that one case I knew I had to make a deepcopy to remove the variants otherwise it would be impossible to have one signal handler that wanted the raw response and another that didn't.

Although as an aside we may want to address that in a future PR. I don't think one signal handler should be able to affect the others.

Reply:
FYI this function is moved to just be a function in `signature.py`. Left this convo unresolved because it seemed like it was still under discussion around the performance bits though